### PR TITLE
[AMD] Fix DLPack Error in Aiter flydsl GEMM by Detaching MoE Gate Weight

### PR DESCRIPTION
**Pull request by bingxche**
## Motivation
The new aiter flydsl GEMM backend exports tensors via DLPack, which raises BufferError on tensors with `requires_grad=True`. MoEGate.weight is an `nn.Parameter` (requires_grad=True by default), causing crashes during CUDA graph capture.
Detach the weight before passing to tgemm.mm. This is a zero-copy operation that shares the same GPU memory, with no precision or performance impact.
Fixes: `nightly-8-gpu-mi35x-kimi-k25` crash

## Accuracy Tests
Before:
image
https://github.com/sgl-project/sglang/actions/runs/23952081831/job/69861755728#step:5:6447

After:
image

https://github.com/sgl-project/sglang/actions/runs/24070615697/job/70206458366#step:5:24477

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

**Review by yctseng0211 — APPROVED** [review]
LGTM

**Review by HaiShaw — APPROVED** [review]
